lib/dataset/bins.cpp

```diff
@@ -215,8 +215,12 @@ Variable lookup_previous(const DataArray &function, const Variable &x, Dim dim,
 Variable pretend_bins_for_threading(const DataArray &da, Dim bin_dim) {
   const auto dim = da.dims().inner();
   const auto size = std::max(scipp::index(1), da.dims()[dim]);
-  // TODO automatic setup with reasonable bin count
-  const auto stride = std::max(scipp::index(1), size / 24);
+  const auto nthread = size > 10000000  ? 24
```

Comment:
Is this the number of tbb threads? I don't see how you can know the number of threads here.

Reply:
True, it is not the number of threads: It is the number of input chunks that is created, which will give the maximum number of threads that could operate on the data.